core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -329,6 +323,7 @@ public void completeWorkflow(Workflow wf) throws Exception {
         workflow.setStatus(WorkflowStatus.COMPLETED);
         workflow.setOutput(wf.getOutput());
         executionDAO.updateWorkflow(workflow);
+        executionDAO.updateTasks(wf.getTasks());
```

Comment:
Same here why do all tasks need to be updated ?

Reply:
In some cases, the tasks are not updated in the DS after completion which causes workflows to be stuck in RUNNING state. This fixes that case.